cfme/tests/configure/test_zones.py

```diff
@@ -14,37 +15,37 @@
 DESC_LEN = 8
 
 
-# Finalizer method that clicks Cancel for tests that expect zone
-# creation to fail. This avoids logging of UnexpectedAlertPresentException
-# for the 'Abandon changes?' alert when the next test tries to navigate
-# elsewhere in the UI.
 def cancel_zone_add(appliance):
+    """Finalizer method that clicks Cancel for tests that expect zone creation to fail.
+    This avoids logging of UnexpectedAlertPresentException for the 'Abandon changes?' alert
+    when the next test tries to navigate elsewhere in the UI."""
+
     view = appliance.browser.create_view(ZoneAddView)
     if view.is_displayed:
         view.cancel_button.click()
 
 
-def create_zone(appliance, name_len=NAME_LEN, desc_len=DESC_LEN):
+def create_zone(appliance, name, desc):
+    """Create zone with the given name and description.
+
+    Returns: :py:class:`cfme.base.Zone` object
+    """
     zc = appliance.collections.zones
     region = appliance.server.zone.region
+    zc.create(name=name, description=desc)
 
-    # CREATE
-    name = fauxfactory.gen_alphanumeric(name_len)
-    description = fauxfactory.gen_alphanumeric(desc_len)
-    zc.create(name=name, description=description)
-
-    # query to get the newly-created zone's id
+    # Re-instantiate the Zone object, to get the id.
     zc.filters = {'parent': region}
     zones = zc.all()
     new_zone = None
     for zone in zones:
-        if (zone.name == name and zone.description == description):
+        if zone.name == name and zone.description == desc:
```

Comment:
*Optional*

Consider implementing an `__eq__` method for the zone class, so that they can be directly compared.

https://github.com/ManageIQ/integration_tests/blob/master/cfme/configure/configuration/region_settings.py#L75